features/netsocket/InternetDatagram.cpp

```diff
@@ -0,0 +1,196 @@
+/* Socket
+ * Copyright (c) 2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "InternetDatagram.h"
+#include "Timer.h"
+#include "mbed_assert.h"
+
+InternetDatagram::InternetDatagram()
```

Comment:
I don't think you can leave the word `Socket` out here. I know `InternetDatagramSocket` is verbose, but it is fundamental. (And most code won't be using that name anyway).